src/opcodes/crypto.cairo

```diff
@@ -238,3 +239,48 @@ pub fn opcode_sha1(ref engine: Engine) -> Result<(), felt252> {
     engine.dstack.push_byte_array(h);
     return Result::Ok(());
 }
+
+pub fn opcode_checksigadd(ref engine: Engine) -> Result<(), felt252> {
+    let pk_bytes = engine.dstack.pop_byte_array()?;
+    let n = engine.dstack.pop_int()?;
+
+    if n > 0x7FFFFFFF || n < -0x80000000 {
+        return Result::Err('InvalidScriptNum');
+    }
+
+    let sig_bytes = engine.dstack.pop_byte_array()?;
+
+    if pk_bytes.len() == 0 {
+        return Result::Err('EmptyPublicKey');
+    }
+
+    let mut is_valid = false;
+
+    if pk_bytes.len() == 32 {
```

Comment:
I think this test can be removed, `verify` result as an error if the key is not in the good format. You're also pushing 33 bytes public key so this test is always false